woocommerce/woocommerce-ios
# TopTabView updates

## Description


- `TopTabView` had a problem where it would always show first tab when created, it uses `selectedTabIndex` as a binding to tell when the selected tab changed, but there was no way to set the initially selected tab
- Using `selectedTabIndex` to set initially selected tab

## Testing information


- Install and set up the Woo Shipping extension on your store.
- Build and run the app with the revampedShippingLabelCreation feature flag enabled.
- Create an order with the processing status and at least one physical product.
- In the order details, select "Create Shipping Label."
- Store options will start fetching from backend in the background.
- Tap "Select a Package" button
- Switch to Carrier tab
- Switch to second carrier in Carrier tab
- Switch to Saved tab
- Go back to Carrier tab and check that UI shows the second carrier

## Screenshots


| Before changes      | After changes |
| ----------- | ----------- |
| ![Simulator Screen Recording - iPhone 16 Pro - 2024-12-04 at 10 29 57](https://github.com/user-attachments/assets/8d252866-4d8c-433d-bb29-067fbc6a0739)     | ![Simulator Screen Recording - iPhone 16 Pro - 2024-12-04 at 10 28 18](https://github.com/user-attachments/assets/28b84204-5445-47dc-9d70-7d831974d127)       |

---

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
